server/util.go

```diff
@@ -324,7 +324,7 @@ func interceptConfigs(rootViper *viper.Viper, customAppTemplate string, customCo
 }
 
 // AddCommands add server commands
-func AddCommands[T types.Application](rootCmd *cobra.Command, appCreator types.AppCreator[T], addStartFlags types.ModuleInitFlags) {
```

Comment:
We add two ways to do the same things for flags: Using `StartCmdOptions` and setting `AddFlags` and then calling `StartCmdWithOptions`. Or use `AddCommands` with custom flags on the start command. This concatenates the possibilities in one.
Note, the backport to v0.50 won't contain that breaking change ( `AddCommandsWithStartCmdOptions` will do what AddCommands does for v0.51)